packages/nestjs/src/setup.ts

```diff
@@ -0,0 +1,156 @@
+import type {
+  ArgumentsHost,
+  CallHandler,
+  DynamicModule,
+  ExecutionContext,
+  NestInterceptor,
+  OnModuleInit,
+} from '@nestjs/common';
+import { Catch } from '@nestjs/common';
+import { Injectable } from '@nestjs/common';
+import { Global, Module } from '@nestjs/common';
+import { APP_FILTER, BaseExceptionFilter } from '@nestjs/core';
+import { APP_INTERCEPTOR } from '@nestjs/core';
+import {
+  SEMANTIC_ATTRIBUTE_SENTRY_OP,
+  SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN,
+  captureException,
+  getClient,
+  getDefaultIsolationScope,
+  getIsolationScope,
+  spanToJSON,
+} from '@sentry/core';
+import type { Span } from '@sentry/types';
+import { logger } from '@sentry/utils';
+import type { Observable } from 'rxjs';
+
+/**
+ * Note: We cannot use @ syntax to add the decorators, so we add them directly below the classes as function wrappers.
+ */
+
+/**
+ * Interceptor to add Sentry tracing capabilities to Nest.js applications.
+ */
+export class SentryTracingInterceptor implements NestInterceptor {
+  /**
+   * Intercepts HTTP requests to set the transaction name for Sentry tracing.
+   */
+  public intercept(context: ExecutionContext, next: CallHandler): Observable<unknown> {
+    if (getIsolationScope() === getDefaultIsolationScope()) {
+      logger.warn('Isolation scope is still the default isolation scope, skipping setting transactionName.');
```

Comment:
Should we hide this log behind a DEBUG_BUILD check?

Reply:
I tried to add this, but it broke some stuff. Noted it down and will look at this after so to not block this PR